src/components/app/data/utils.js

```diff
@@ -747,3 +747,54 @@ export function isEnrollmentUpgradeable(enrollment) {
   const canUpgradeToVerifiedEnrollment = enrollment.mode === COURSE_MODES_MAP.AUDIT && !isEnrollByLapsed;
   return canUpgradeToVerifiedEnrollment;
 }
+
+export function determineAllocatedCourseRuns({
```

Comment:
nit: perhaps considering renaming to `determineAllocatedCourseRunAssignmentsForCourse` in order to be explicit that:
1. this function is dealing with allocated _assignments_. (e.g., otherwise, could this be referring to allocated subscription license, etc.?).
2. mention of "ForCourse" (or otherwise) in the function name helps declare its intent a bit more explicitly, that it's not just allocated course run assignments, but specifically course run assignments _for a specific course_.